:book: add note about officially versions supported (#1735)
kubernetes-sigs/kubebuilder: docs/book/src/quick-start.md

- [go](https://golang.org/dl/) version v1.15+.
- [docker](https://docs.docker.com/install/) version 17.03+.
- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.11.3+.
- Access to a Kubernetes v1.11.3+ cluster.

<aside class="note">
<h1>Versions</h1>

> Kubebuilder will create a project for you where the officially versions supported are defined on its Makefile and go.mod file. For example,  see that it will install a specific version for Kustomize which is a prerequisite for the tool.  

Reviewer: Can you make a list of what these binaries are?

Author: would not it be too verbose and may outdated easily? How do you think that it should be? Could you please give an example?

Reviewer: We should be listing project dependencies, and updating that list when we make changes. We don't add/remove tools often so I don't see this list getting outdated quickly.
```suggestion
Projects created by kubebuilder contain a Makefile that will install tools at versions defined at creation time. Those tools are:
- [kustomize](https://kubernetes-sigs.github.io/kustomize/)
- [controller-gen](https://github.com/kubernetes-sigs/controller-tools)
```

Author: Done: 

<img width="781" alt="Screen Shot 2020-10-22 at 16 08 28" src="https://user-images.githubusercontent.com/7708031/96918548-5fd53d80-14a2-11eb-9a7d-b78ee64172dc.png">

Thanks for your feedback.

- [go](https://golang.org/dl/) version v1.15+.
- [docker](https://docs.docker.com/install/) version 17.03+.
- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.11.3+.
- Access to a Kubernetes v1.11.3+ cluster.

<aside class="note">
<h1>Versions and Supportability</h1>

Projects created by Kubebuilder contain a Makefile that will install tools at versions defined at creation time. Those tools are:
- [kustomize](https://kubernetes-sigs.github.io/kustomize/)
- [controller-gen](https://github.com/kubernetes-sigs/controller-tools)

> Note that, the officially supported versions are that ones defined in the Makefile and go.mod files which are generated by tool. 

Reviewer: No `go.mod` entry is created for any tool since they're installed in temp modules outside of a project tree
```suggestion
We test with and therefore recommend using the defined versions.
```

Author: The pronoun `we` should be avoided. See: https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we

Also, I'd like to make clear that the versions on go.mod and in the Makefile are the supported ones. I mean, users are able to customize it as they wish, however, we are unable to ensure the behaviour with other versions. I tried to address your suggestion and keep the idea/motivation. Let's see. Please, feel free to make a new suggestion if you wish as well.
